### PR TITLE
conditional compilation for net 8

### LINKED ISSUE #2111: [Feature Request] Conditional compilation for net8.0
As of #2108, net8.0 is a required build target. Add a condition to the build props that targets net8.0 if an environment variable is set, making it opt-in.

### PR DESCRIPTION
**Pull request by westin-m**
opt in for targeting net 8 through env variable

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -42,8 +42,15 @@
     <NoWarn>$(NoWarn);SYSLIB0050</NoWarn>
     <NoWarn>$(NoWarn);SYSLIB0051</NoWarn>
   </PropertyGroup>
-
-  <ItemGroup>
+  
+  <PropertyGroup>
+    <EnableMicrosoftExtensionsConfigurationBinderSourceGenerator>
+      true
```
Comment (jmprieur) [inline build/common.props]: Does Wilson use `Configuration.Bind`?
This is for Id.Web, not sure for Wilson!

**Review by jennyf19 — APPROVED** [review]
LGTM....will you do something similar in Id Web?

**Review by jmprieur — APPROVED** [review]
Thanks @westin-m 
I checked, this build on a machine where  .NET 8 is not installed!

File: build/common.props (hunk 1 above)
Comment (keegan-caruso) [reply build/common.props]: same question, I don't see why this is needed

**Pull request merged**